search/includes/classes/commands/class-corecommand.php

```diff
@@ -40,7 +35,13 @@ protected function _maybe_setup_index_version( &$assoc_args ) {
 					return WP_CLI::error( sprintf( 'Indexable %s not found - is the feature active?' ) );
 				}
 
-				$result = $search->versioning->set_current_version_number( $indexable, $version );
+				$new_version_number = $search->versioning->normalize_version_number( $indexable, $desired_version_number );
```

Comment:
This is normalizing the version only for the main site, so if `next` is 2 for the main site, it will index version 2 for all sites. It should be done inside the `foreach ( $sites as $site )` loop. The problem is that loop is in `Command::index()` and I guess you were trying to avoid tampering with versions in that file... 😕

Reply:
You're totally right, whoops. Good catch!

Reply:
@netsuso I believe I have addressed this in ad99b5200, do you mind checking that? Basically what I did was _not_ normalize the index version here and inside Versioning::set_current_version_number(), so that we can store the alias there (`next`) and dynamically translate that to an actual version number at runtime.

I also remove the conditional that used `get_site_option()` in network mode - that will affect prod multisites that have index versions, so we'll need to proceed carefully for deployment.

Reply:
Fixed the `index --network-wide` command, thanks! I'm glad `current_index_version_by_type` existed, it made things easier than I expected 😊

As for the `get_site_option()`, I think I'll wait until your morning to deploy this, as I'm not 100% sure about its implications for multisites and what can happen if no version option is available. IIRC it returns 1 by default, but that could be a problem if another version had been previously generated... Although I suppose we would have noticed earlier in this case.